File: Knjigoteka.Model/Entities/Branch.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Knjigoteka.Model.Entities
{
    public class Branch
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; } = null!;
        [Required]
        public int CityId { get; set; }
        [Required]
        public string Address { get; set; } = null!;
        [Required]
        [Phone(ErrorMessage = "Invalid phone number format")]
        public string PhoneNumber { get; set; } = null!;
        [Required]
        public TimeOnly OpeningTime { get; set; }
        [Required]
        public TimeOnly ClosingTime { get; set; }
        [ForeignKey(nameof(CityId))]
        public City City { get; set; } = null!;
        public ICollection<BookBranch> BookBranches { get; set; } = new List<BookBranch>();
        public ICollection<Employee> Employees { get; set; } = new List<Employee>();
        public ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();
        public ICollection<RestockRequest> RestockRequests { get; set; } = new List<RestockRequest>();

    }
}

File: Knjigoteka.UI/knjigoteka_desktop/lib/models/branch.dart
class Branch {
  final int id;
  final String name;
  final int cityId;
  final String cityName;
  final String address;
  final String phoneNumber;
  final String openingTime;
  final String closingTime;

  Branch({
    required this.id,
    required this.name,
    required this.cityId,
    required this.cityName,
    required this.address,
    required this.phoneNumber,
    required this.openingTime,
    required this.closingTime,
  });

  factory Branch.fromJson(Map<String, dynamic> json) => Branch(
    id: json['id'],
    name: json['name'],
    cityId: json['cityId'],
    cityName: json['cityName'],
    address: json['address'],
    phoneNumber: json['phoneNumber'],
    openingTime: json['openingTime'],
    closingTime: json['closingTime'],
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'cityId': cityId,
    'address': address,
    'phoneNumber': phoneNumber,
    'openingTime': openingTime,
    'closingTime': closingTime,
  };
}
